tla-io/src/main/scala/at/forsyte/apalache/tla/imp/SanyImporter.scala

```diff
@@ -106,34 +106,41 @@ class SanyImporter(sourceStore: SourceStore, annotationStore: AnnotationStore) e
    * into it, in order to call SANY.
    *
    * @param source
-   *   the text source
+   *   the text source for the root module
+   * @param aux
+   *   the text sources for any auxiliary modules
    * @return
    *   the pair (the root module name, a map of modules)
    */
-  def loadFromSource(source: Source): (String, Map[String, TlaModule]) = {
+  def loadFromSource(source: Source, aux: Seq[Source] = Seq()): (String, Map[String, TlaModule]) = {
+    val tempDir = Files.createTempDirectory("sanyimp").toFile
+    try {
+      val rootModule = saveTlaFile(tempDir, source)
+      aux.foreach(saveTlaFile(tempDir, _))
```

Comment:
Do we need any error handling for when the same file is passed multiple times?

Reply:
Maybe there should be a test for that.

Reply:
Added a check for this in fe66642038af698a4809607988860bb653d40f8d